### DISCUSSION TITLE
Tools for Creating Animated Interactive Charts with AI (No Programming Experience)

### DISCUSSION THREAD

**Discussion by iposta** [community/community#140479] [Programming Help] [answered]
### Body

I want to create animated and interactive charts using AI coding.

I have no programming knowledge, and it would be amazing if I could easily run and preview the code generated by AI, similar to how Claude's preview window works.

My goal is to describe what I want entirely in text and ideally get a result that can be played in the browser. For example, I might ask for a 4-column chart with specific values, where initially only the values are shown, and when I click on each value, its corresponding bar animates into view using a scale animation. Or even more complex variations of this concept.

What tools or platforms can help me achieve this?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Topromm** [comment, marked as answer]
**ChatGPT with Code Interpreter**
- Describe your chart, and ChatGPT generates Python/JavaScript code.
- You can run and preview the code directly with the interpreter.

**Google Colab**
- Use AI-generated Python code (e.g., Plotly) to run and view charts in your browser.

**Observable**
- No-code platform for creating interactive charts using D3.js, with easy-to-use templates.

**ChartBlocks**
- A drag-and-drop chart creation tool that supports some level of interactivity.

I think it's a case of doing a lot of trial and error and trying out different things. Describing what you want in a way that is understandable to the machine can be tough and knowing some related terms would certainly not hurt. You could also try chopping it into smaller tasks and describe how to animate it in the end. Best of luck!

**Reply by iposta** [reply]
An amazing response, Topias. No amount of thanks would be enough for you.

**Reply by ebndev** [reply]
@iposta please feel free to come back to Discussions and share when you've built this :)

**Reply by Mti-isf** [reply]
mkoko143125@gmail.com

**Comment by AbdulAhad365** [comment]
Here are three platforms for creating animated, interactive charts without coding skills:

ObservableHQ:

Browser-based, interactive chart creation using pre-built templates.
[observablehq.com](https://observablehq.com/)
Glide Apps:

Build interactive apps with data visualizations from Google Sheets.
[glideapps.com](https://www.glideapps.com/)
OpenAI Code Interpreter:

Use AI to generate chart code from text descriptions and preview it in the browser.

**Reply by iposta** [reply]
Thank you very much
